runtime/src/comm/ugni/comm-ugni.c

```diff
@@ -5285,6 +5292,141 @@ void do_remote_put_V(int v_len, void** src_addr_v, c_nodeid_t* locale_v,
 }
 
 
+static
+void do_remote_get_V(int v_len,
+                     void** tgt_addr_v, c_nodeid_t* locale_v, mem_region_t** remote_mr_v,
+                     void** src_addr_v, size_t* size_v, mem_region_t** local_mr_v) {
+
+  DBG_P_LP(DBGF_GETPUT, "DoRemGetV(%d) %p -> %d:%p (%#zx), proxy %c",
+           v_len, src_addr_v[0], (int) locale_v[0], tgt_addr_v[0], size_v[0],
+           may_proxy ? 'y' : 'n');
+
+#if HAVE_GNI_FMA_CHAIN_TRANSACTIONS
+
+  //
+  // This GNI is new enough to support chained transactions.
+  //
+
+  //
+  // If there are more than we can handle at once, block them up.
+  //
+  while (v_len > MAX_CHAINED_GET_LEN) {
+    do_remote_get_V(MAX_CHAINED_GET_LEN, tgt_addr_v, locale_v, remote_mr_v,
+                    src_addr_v, size_v, local_mr_v);
+    v_len -= MAX_CHAINED_GET_LEN;
+    src_addr_v += MAX_CHAINED_GET_LEN;
+    locale_v += MAX_CHAINED_GET_LEN;
+    tgt_addr_v += MAX_CHAINED_GET_LEN;
+    size_v += MAX_CHAINED_GET_LEN;
+    local_mr_v  += MAX_CHAINED_GET_LEN;
+    remote_mr_v += MAX_CHAINED_GET_LEN;
+  }
+
+  if (v_len <= 0)
+    return;
+
+  //
+  // Do all these GETs in one chained transaction.  Except: if we have to
```

Comment:
The logic to achieve this (scalar PUT if unregistered) is somewhere else rather than here, isn't it?  The comment ought to be wherever that is.

Reply:
It doesn't exist currently (one of the listed TODOs) but it will go around here